src/Juvix/Data/Effect/Forcing.hs

```diff
@@ -0,0 +1,20 @@
+{-# OPTIONS_GHC -Wno-unused-type-patterns #-}
+
+module Juvix.Data.Effect.Forcing where
+
+import Juvix.Prelude.Base
+
+data Forcing (a :: GHCType) :: Effect where
+  ForcesField :: (NFData b) => Lens' a b -> Forcing a m ()
+  Forces :: Lens' a b -> Sem '[Forcing b] () -> Forcing a m ()
+
+makeSem ''Forcing
+
+forcing :: a -> Sem '[Forcing a] () -> a
```

Comment:
What does forcing do? Perhaps write a docstring to explain?

Reply:
it is a special case of `evalForcing` for `r = []`. I added this because I think it will be common enough to use the `Forcing` effect on its own.

Reply:
What does the `Forcing` effect do?

Reply:
I've added documentation. Please let me know if it needs more explanation